api/mapping/services_nlp.py

```diff
@@ -1,19 +1,305 @@
+        codes_dict = concept_code_to_id(codes)
+
+        # Mini function to see if all conceptIDs are the same
+        def all_same(items):
+            return all(x == items[0] for x in items)
+
+        # Check each item in values and see whether NLP got a result
+        # If NLP finds something, save the result to ScanReportConcept
+        # If all conceptIDs across vocabs are the same, save only SNOMED
+        # Else save each conceptID to ScanReportConcept
+        for value in scan_report_values:
+            match = list(
+                filter(lambda item: item["pk"] == str(value.id), codes_dict))
+            print('MATCH >>> ', match)
+            concept_ids = [li['conceptid'] for li in match]
+            print('CONCEPT IDs >>> ', concept_ids)
+            print('ALL SAME >>> ', all_same(concept_ids))
+
+            # If there are multiple conceptIDs from the above filter
+            if len(concept_ids) > 0:
+
+                # If all conceptIDs are the same, grab only the SNOMED entry
+                # and save this to ScanReportConcept
+                if all_same(concept_ids):
+                    # Grab the SNOMED dictionary element
+                    same = list(
+                        filter(lambda item: item['nlp_vocab'] == 'SNOMEDCT_US', match))
+                    scan_report_value = ScanReportValue.objects.get(
+                        pk=same[0]["pk"])
+                    concept = Concept.objects.get(pk=same[0]["conceptid"])
+
+                    ScanReportConcept.objects.create(
+                        nlp_entity=same[0]["nlp_entity"],
+                        nlp_entity_type=same[0]["nlp_entity_type"],
+                        nlp_confidence=same[0]["nlp_confidence"],
+                        nlp_vocabulary=same[0]["nlp_vocab"],
+                        nlp_concept_code=same[0]["nlp_code"],
+                        concept=concept,
+                        content_object=scan_report_value,
+                    )
+
+                else:
+
+                    # If the conceptIDs are all different then save each to ScanReportConcept
+                    for item in match:
+                        print('ITEM >>> ', item)
+                        scan_report_value = ScanReportValue.objects.get(
+                            pk=item["pk"])
+                        concept = Concept.objects.get(pk=item["conceptid"])
+
+                        ScanReportConcept.objects.create(
+                            nlp_entity=item["nlp_entity"],
+                            nlp_entity_type=item["nlp_entity_type"],
+                            nlp_confidence=item["nlp_confidence"],
+                            nlp_vocabulary=item["nlp_vocab"],
+                            nlp_concept_code=item["nlp_code"],
+                            concept=concept,
+                            content_object=scan_report_value,
+                        )
+
+    return True
+
+
+def nlp_single_string(pk, dict_string):
```

Comment:
This function name could reflect what happens in it. Example: get_concept_id_list_from_nlp_string?

Reply:
I've deleted this function because we no longer need it now NLP is integrated at the field/value level.